src/api/gnosisProtocol/hooks.ts

```diff
@@ -12,6 +12,7 @@ import { isBarnBackendEnv } from 'legacy/utils/environments'
 import { isTruthy } from 'legacy/utils/misc'
 import { supportedChainId } from 'legacy/utils/supportedChainId'
 
+import { emulatedPartOrdersAtom } from 'modules/twap/state/emulatedPartOrdersAtom'
```

Comment:
Yeah, it makes my life harder :(
We have a dependency inversion principle breach here.
There is a TODO to fix it:
```
/**
 * TODO: refactor this hook
 * Currently there is a problem with Dependency inversion principle
 * This hook depends on 'modules/twap', but it must not
 *
 * This violation is added because of environment restrictions for TWAP
 * Currently, WatchTower creates discrete orders only on PROD environment
 * And even if you created a TWAP order not in PROD, we have to load discrete orders from PROD order-book
 *
 * For non-PROD environment we do two requests: barn and prod
 * For PROD environment we do only one: prod. It depends on isBarnBackendEnv
 */
```